### ISSUE TITLE
Is the gridding operation differentiable?

### ISSUE THREAD

**Issue by Michaelsqj** [tomer196/PILOT#13]
Hi Tomer,

Thanks for sharing your code! I noticed you implemented gridding https://github.com/tomer196/PILOT/blob/9530139d9fbacda064d11fc12614ea4a6a74b7f6/pytorch_nufft/interp.py#L84
May I ask can this function backpropogate gradient? 

I have some parameters that controls a non-Cartesian trajectory, and I hope to grid the non-Cartesian trajectory onto Cartesian grid, so I could have a rough density in different regions of k-space and then derive a cost function based on the value in Cartesian grid to optimise the control parameters.  

I'm wondering if I could use your `_gridding2()` function to achieve that?

Yours,
Qijia

**Comment by tomer196** [comment]
Yes, this is exactly what we did in this work. We subsampled the data using the NUFFT and then used a reconstruction network. The gradients can backpropagate all the way back to the trajectory coordinates.
If you have any other question feel free to ask.
